### PR TITLE
Checkuser

### PR DESCRIPTION
**Pull request by olusakinTim**
checking if users exist

### CODE REVIEW

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs [hunk 1]
```diff
@@ -20,7 +23,106 @@ public SectorController(IConfiguration _config, SIWPDBContext context)
...
+        [Route("api/Project/DeleteSector")]
+        public CreateSectorResponses DeleteSector(SectorPayload sectorinfo)
+        {
+            try
+            {
+                using (var db = _context)
+                {
+                    var sector = db.TbSectors.Find(sectorinfo.SectorId);
+                    if (sector == null)
+                    {
```
Comment (toniyide) [inline SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs]: the route on the controllers are not corresponding with the controller name and the method name

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Models/TbOrganisations.cs [hunk 2]
```diff
@@ -14,7 +14,9 @@ public TbOrganisations()
         public string Name { get; set; }
         public string LongName { get; set; }
         public int? OrgTypeId { get; set; }
-        public string Status { get; set; }
+        public string Status {
+            get{ return Status; }
+            set{ Status =  "active"; } }
```
[the remark this answers is not in the document]
Comment (olusakinTim) [reply SIWP_Revamp_Project/SIWP-Revamp-Main/Models/TbOrganisations.cs]: guess i forgot to delete it creating another pull request now

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs [hunk 3]
```diff
@@ -20,7 +23,106 @@ public SectorController(IConfiguration _config, SIWPDBContext context)
...
+                    return (new CreateSectorResponses { status = true, message = sJSONResponse});
+                }
+            }
+            catch(Exception e)
+            {
+               new LogWriter(e.Message);
+               return (new CreateSectorResponses { status = false, message = "Couldn't get the rows" });
+            }
+        }
     }
```
Comment (toniyide) [inline SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs]: send a list of the sectors. don't serialize it.

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Responses/CreateSectorResponses.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+using System.Threading.Tasks;
+
+namespace SIWP_Revamp_Main
+{
+    public class CreateSectorResponses
+    {
+        public bool status { get; set; }
+        public string message { get; set; }
+    }
+}
```
Comment (toniyide) [inline SIWP_Revamp_Project/SIWP-Revamp-Main/Responses/CreateSectorResponses.cs]: your CreateSectorResponses  has two fields status and message .. why then are you returning an object of  CreateSectorResponses  in the GetAllSectors api

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Utilities/Encryptor.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                chars[i] = allowedChars[Convert.ToInt32((allowedChars.Length) * randNum.NextDouble())];
+            }
+            return new string(chars);
+        }
+        public static string EncodePassword(string pass, string salt) //encrypt password    
+        {
+            byte[] bytes = Encoding.Unicode.GetBytes(pass);
+            byte[] src = Encoding.Unicode.GetBytes(salt);
+            byte[] dst = new byte[src.Length + bytes.Length];
+            System.Buffer.BlockCopy(src, 0, dst, 0, src.Length);
```
Comment (toniyide) [inline SIWP_Revamp_Project/SIWP-Revamp-Main/Utilities/Encryptor.cs]: what did you change here?

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs (hunk 3 above)
Comment (olusakinTim) [reply SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs]: tried that it was given errors 
wanted to see you today about it which was better

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Utilities/Encryptor.cs (hunk 5 above)
Comment (olusakinTim) [reply SIWP_Revamp_Project/SIWP-Revamp-Main/Utilities/Encryptor.cs]: Nothing

File: SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs (hunk 1 above)
Comment (olusakinTim) [reply SIWP_Revamp_Project/SIWP-Revamp-Main/Controllers/SectorController.cs]: ooh i get

**Pull request closed without merging**